runtime/vm/jvminit.c

```diff
@@ -7283,7 +7283,7 @@ protectedInitializeJavaVM(J9PortLibrary* portLibrary, void * userData)
 		IDATA disabled = FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXPRINTFLAGSFINALDISABLE, NULL);
 		if (enabled > disabled) {
 			size_t maxHeapSize = (size_t) (vm->memoryManagerFunctions->j9gc_get_maximum_heap_size(vm));
-			uint64_t maxDirectMemorySize = (uint64_t) (vm->directByteBufferMemoryMax);
+			uint64_t maxDirectMemorySize = (uint64_t) (vm->directByteBufferMemoryMax == (UDATA)-1 ? 0 : vm->directByteBufferMemoryMax);
```

Comment:
A better way to say `(DATA)-1` is `~(UDATA)0` (the former assumes sufficient sign-extension while the latter does not). Also, there should be parentheses around the `==` sub-expression.

Reply:
Updated.